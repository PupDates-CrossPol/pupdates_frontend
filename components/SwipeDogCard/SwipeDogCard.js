import * as React from 'react'
import { StyleSheet, View, Image, TouchableOpacity} from 'react-native';
import { connect } from 'react-redux'
import { SliderBox } from "react-native-image-slider-box";
import { Container, Header, Content, Card, CardItem, Text, Body, Button } from 'native-base';
import { Ionicons } from '@expo/vector-icons';



export const SwipeDogCard = (props) => {
  const dogCards = props.swipePack.map((dog, index) => {
    const currentImages = props.swipePackPhotos.filter(pic => pic.dog_id === dog.id)
    const boxPics = currentImages.map(image => image.image_url)
    return (
      <Card style={styles.imageCard}>
        {/* <Text style={styles.packName}>{props.swipeUser.attributes.first_name}'s Pack</Text> */}
          <CardItem style={styles.imageCardContent}>
            <Body>
            <SliderBox images={boxPics} style={styles.image} dotColor='rgb(21, 112, 125)' circleLoop />
            </Body>
          </CardItem>
          <CardItem style={styles.imageCardContentText}>
            <Body style={styles.infoBody}>
              <Text style={styles.infoTextName}>{dog.name}</Text>
              <Text style={styles.infoText}>Breed: {dog.breed}</Text>
              <Text style={styles.infoText}>Age: {dog.age}</Text>
              <Text style={styles.infoText}>Gender: {dog.sex}</Text>
              <Text style={styles.infoText}>Fixed: {dog.fixed ? 'true' : 'false'}</Text>
              <Text style={styles.infoText}>Vaccinated: {dog.vaccinated ? 'true' : 'false'}</Text>
              <Text style={styles.infoText}>Good with kids: {dog.good_with_kids ? 'true' : 'false'}</Text>
            </Body>
          </CardItem>
        </Card> 
    )
   
    
  })
  return (
      <>
        {dogCards}
      </>
  )}



  const styles = StyleSheet.create({
    container: {
      flex: 1,
      backgroundColor: '#fff',
      alignItems: 'center',
      // marginTop: 90
    },
    logo: {
      height: 40,
      width: 40,
    },
    title: {
      fontSize: 50,
    },
    imageCard: {
      flexDirection: 'column',
      alignItems: 'center',
      justifyContent: 'center',
      marginTop: 15,
    },
    infoCard: {
      flexDirection: 'column',
      alignItems: 'center',
      justifyContent: 'center',
      marginTop: 10,
      width: '80%'
    },
    packName: {
      fontSize: 25
    },
    imageCardContent: {
      alignItems: 'center',
      justifyContent: 'center',
      flexDirection: 'column',
      height: 350,
    },
  
    imageCardContentText: {
      alignItems: 'center',
      justifyContent: 'center',
      flexDirection: 'column',
      height: 250,
      width: '80%'
    },
    buttonText: {
      color: '#fff',
      fontSize: 25,
    },
    navTitle: {
      width: 160,
      height: 40,
      marginBottom: 5,
  },
  rightNavIcon: { 
      marginRight: 10,  
  },
  leftNavIcon: { 
    marginLeft: 10,  
    marginBottom: 5, 
  },
    image: {
      height: 325,
      width: 275,
      alignItems: 'center',
      justifyContent: 'center',
      borderColor: 'black',
      borderWidth: 3.5,
      borderRadius: 50,
      alignSelf: 'center'
    },
    infoText: {
      textAlign: 'left',
      fontSize: 20,
      color: 'grey',
      alignSelf: 'stretch'
    },
    infoTextName: {
      fontSize: 30,
      textAlign: 'left'
    },
    infoBody: {
      // alignItems: 'center',
      flexDirection: 'column',
      justifyContent: 'center',
      textAlign: 'left',
      height: 50
    },
    roundBtn: {
      width: 40
    },
    pawBtn: {
      alignItems: 'flex-end',
      flexDirection: 'row',
      justifyContent: 'space-around',
      width: '90%',
    }
  });
  


export const mapStateToProps = (state) => ({
  swipePack: state.swipePack,
  swipePackPhotos: state.swipePackPhotos
})

export default connect (mapStateToProps)(SwipeDogCard)






