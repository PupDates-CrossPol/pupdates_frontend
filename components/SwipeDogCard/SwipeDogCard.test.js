import React from 'react'
import renderer from 'react-test-renderer'
import { SwipeDogCard } from './SwipeDogCard'
import { Provider } from 'react-redux'

describe('<SwipeDogCard />', () => {
  it('has one child', () => {
    const mockSwipePack = [
      {
        id: 1,
        user_id: 1,
        name: 'fluffy'
      }
    ]
    const mockSwipePackPhotos = [
      {
        id: 1,
        dog_id: 1,
        image_url: 'woof'
      }
    ]
    const tree = renderer.create(<SwipeDogCard swipePack={mockSwipePack} swipePackPhotos={mockSwipePackPhotos} />).toJSON()
    expect(tree.children.length).toBe(2)
  })
  it('renders correctly', () => {
    const mockSwipePack = [
      {
        id: 1,
        user_id: 1,
        name: 'fluffy'
      }
    ]
    const mockSwipePackPhotos = [
      {
        id: 1,
        dog_id: 1,
        image_url: 'woof'
      }
    ]
    const tree = renderer.create(<SwipeDogCard swipePack={mockSwipePack} swipePackPhotos={mockSwipePackPhotos} />).toJSON()

    expect(tree).toMatchSnapshot()
  })
})