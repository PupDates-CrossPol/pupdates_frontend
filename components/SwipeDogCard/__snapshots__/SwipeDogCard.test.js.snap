// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SwipeDogCard /> renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#fff",
        "borderBottomWidth": 0.5,
        "borderColor": "#ccc",
        "borderLeftWidth": 0.5,
        "borderRadius": 2,
        "borderRightWidth": 0.5,
        "borderTopWidth": 0.5,
        "elevation": 3,
        "flexWrap": "nowrap",
        "marginBottom": 5,
        "marginLeft": 2,
        "marginRight": 2,
        "marginTop": 5,
        "shadowColor": "#000",
        "shadowOffset": Object {
          "height": 2,
          "width": 0,
        },
        "shadowOpacity": 0.1,
        "shadowRadius": 1.5,
      },
      Object {
        "alignItems": "center",
        "flexDirection": "column",
        "justifyContent": "center",
        "marginTop": 15,
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#fff",
          "borderRadius": 2,
          "flexDirection": "row",
          "paddingBottom": 10,
          "paddingLeft": 15,
          "paddingRight": 15,
          "paddingTop": 10,
        },
        Object {
          "alignItems": "center",
          "flexDirection": "column",
          "height": 350,
          "justifyContent": "center",
        },
      ]
    }
  >
    <View
      style={
        Object {
          "alignItems": "flex-start",
          "alignSelf": "stretch",
          "flex": 1,
        }
      }
    >
      <View
        style={
          Object {
            "borderRadius": 2,
          }
        }
      >
        <RCTScrollView
          ImageComponent={[Function]}
          activeAnimationOptions={null}
          activeAnimationType="timing"
          activeSlideAlignment="center"
          activeSlideOffset={20}
          apparitionDelay={0}
          automaticallyAdjustContentInsets={false}
          autoplay={false}
          autoplayDelay={1000}
          autoplayInterval={3000}
          borderTopLeftRadius={2}
          borderTopRightRadius={2}
          callbackOffsetMargin={5}
          circleLoop={true}
          containerCustomStyle={Object {}}
          contentContainerCustomStyle={Object {}}
          contentContainerStyle={
            Array [
              Object {
                "paddingLeft": 0,
                "paddingRight": 0,
              },
              Object {},
            ]
          }
          data={
            Array [
              "woof",
            ]
          }
          decelerationRate="fast"
          directionalLockEnabled={true}
          disableVirtualization={false}
          dotColor="rgb(21, 112, 125)"
          enableMomentum={false}
          enableSnap={true}
          firstItem={0}
          getItem={[Function]}
          getItemCount={[Function]}
          hasParallaxImages={false}
          horizontal={true}
          images={
            Array [
              "woof",
            ]
          }
          inactiveSlideOpacity={0.7}
          inactiveSlideScale={0.9}
          inactiveSlideShift={0}
          initialNumToRender={6}
          invertStickyHeaders={false}
          inverted={false}
          itemWidth={750}
          keyExtractor={[Function]}
          layout="default"
          lockScrollTimeoutDuration={1000}
          lockScrollWhileSnapping={false}
          loop={true}
          loopClonesPerSide={5}
          maxToRenderPerBatch={13}
          numColumns={1}
          onContentSizeChange={[Function]}
          onEndReachedThreshold={2}
          onLayout={[Function]}
          onMomentumScrollEnd={[Function]}
          onResponderRelease={[Function]}
          onScroll={[Function]}
          onScrollBeginDrag={[Function]}
          onScrollEndDrag={[Function]}
          onSnapToItem={[Function]}
          onStartShouldSetResponderCapture={[Function]}
          onTouchEnd={[Function]}
          onTouchStart={[Function]}
          overScrollMode="never"
          pinchGestureEnabled={false}
          removeClippedSubviews={true}
          renderItem={[Function]}
          scrollEnabled={true}
          scrollEventThrottle={1}
          scrollsToTop={false}
          shouldOptimizeUpdates={true}
          showsHorizontalScrollIndicator={false}
          showsVerticalScrollIndicator={false}
          slideStyle={Object {}}
          sliderWidth={750}
          stickyHeaderIndices={Array []}
          style={
            Object {
              "flexDirection": "row",
              "opacity": 0,
              "width": 750,
            }
          }
          swipeThreshold={20}
          updateCellsBatchingPeriod={50}
          useScrollView={false}
          vertical={false}
          viewabilityConfigCallbackPairs={Array []}
          windowSize={13}
        >
          <View>
            <View
              onLayout={[Function]}
              style={
                Array [
                  Object {
                    "flexDirection": "row",
                  },
                  null,
                ]
              }
            >
              <View
                pointerEvents="box-none"
                style={
                  Object {
                    "opacity": 1,
                    "transform": Array [
                      Object {
                        "scale": 1,
                      },
                    ],
                    "width": 750,
                  }
                }
              >
                <View
                  accessible={true}
                  focusable={true}
                  isTVSelectable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={null}
                >
                  <Image
                    ImageComponent={[Function]}
                    circleLoop={true}
                    dotColor="rgb(21, 112, 125)"
                    images={
                      Array [
                        "woof",
                      ]
                    }
                    resizeMethod="resize"
                    resizeMode="cover"
                    source={
                      Object {
                        "uri": "woof",
                      }
                    }
                    style={
                      Object {
                        "alignItems": "center",
                        "alignSelf": "center",
                        "borderColor": "black",
                        "borderRadius": 50,
                        "borderWidth": 3.5,
                        "height": 325,
                        "justifyContent": "center",
                        "width": 275,
                      }
                    }
                  />
                </View>
              </View>
            </View>
          </View>
        </RCTScrollView>
      </View>
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#fff",
          "borderRadius": 2,
          "flexDirection": "row",
          "paddingBottom": 10,
          "paddingLeft": 15,
          "paddingRight": 15,
          "paddingTop": 10,
        },
        Object {
          "alignItems": "center",
          "flexDirection": "column",
          "height": 250,
          "justifyContent": "center",
          "width": "80%",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "flex-start",
            "alignSelf": "stretch",
            "flex": 1,
          },
          Object {
            "flexDirection": "column",
            "height": 50,
            "justifyContent": "center",
            "textAlign": "left",
          },
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#000",
              "fontFamily": "System",
              "fontSize": 16,
            },
            Object {
              "fontSize": 30,
              "textAlign": "left",
            },
          ]
        }
        uppercase={false}
      >
        fluffy
      </Text>
      <Text
        style={
          Array [
            Object {
              "color": "#000",
              "fontFamily": "System",
              "fontSize": 16,
            },
            Object {
              "alignSelf": "stretch",
              "color": "grey",
              "fontSize": 20,
              "textAlign": "left",
            },
          ]
        }
        uppercase={false}
      >
        Breed: 
      </Text>
      <Text
        style={
          Array [
            Object {
              "color": "#000",
              "fontFamily": "System",
              "fontSize": 16,
            },
            Object {
              "alignSelf": "stretch",
              "color": "grey",
              "fontSize": 20,
              "textAlign": "left",
            },
          ]
        }
        uppercase={false}
      >
        Age: 
      </Text>
      <Text
        style={
          Array [
            Object {
              "color": "#000",
              "fontFamily": "System",
              "fontSize": 16,
            },
            Object {
              "alignSelf": "stretch",
              "color": "grey",
              "fontSize": 20,
              "textAlign": "left",
            },
          ]
        }
        uppercase={false}
      >
        Gender: 
      </Text>
      <Text
        style={
          Array [
            Object {
              "color": "#000",
              "fontFamily": "System",
              "fontSize": 16,
            },
            Object {
              "alignSelf": "stretch",
              "color": "grey",
              "fontSize": 20,
              "textAlign": "left",
            },
          ]
        }
        uppercase={false}
      >
        Fixed: 
        false
      </Text>
      <Text
        style={
          Array [
            Object {
              "color": "#000",
              "fontFamily": "System",
              "fontSize": 16,
            },
            Object {
              "alignSelf": "stretch",
              "color": "grey",
              "fontSize": 20,
              "textAlign": "left",
            },
          ]
        }
        uppercase={false}
      >
        Vaccinated: 
        false
      </Text>
      <Text
        style={
          Array [
            Object {
              "color": "#000",
              "fontFamily": "System",
              "fontSize": 16,
            },
            Object {
              "alignSelf": "stretch",
              "color": "grey",
              "fontSize": 20,
              "textAlign": "left",
            },
          ]
        }
        uppercase={false}
      >
        Good with kids: 
        false
      </Text>
    </View>
  </View>
</View>
`;
